File: Backend/src/main/java/com/app/distrifoods/app/entities/Usuario.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.app.distrifoods.app.entities;

import java.io.Serializable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name="usuarios")
public class Usuario implements Serializable{
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name="id_usuario")
    private Integer id;
    @Column(name="identificacion", unique=true)
    private Long identificacion;
    @Column(name="nom_usuario", nullable=false)
    private String nombre;
    @Column(name="contrasena", nullable=false)
    private String contrasena;
    @Column(name="correo", nullable=false, unique=true)
    private String correo;
}

File: Frontend/rotonda-front/src/app/models/usuario.ts
import { Cliente } from "./cliente";

export interface Usuario {
  id:  number;
  identificacion: number;
  nombre: string;
  contrasena: string;
  correo: string;
}


export interface UsuarioLogueadoResponse {
  cliente:  Cliente;
  identificacion: number;
  nombre: string;
  usuario: string;
  contrasena: string;
}

export interface UsuarioLogueadoRequest {
  identificacion: number;
  contrasena: string;
}

